src/wymeditor/core.js

```diff
@@ -541,251 +831,39 @@ jQuery.extend(WYMeditor, {
 
     `jQuery(".wymeditor").wymeditor({});`
 */
-jQuery.fn.wymeditor = function (options) {
-    var $textareas = this;
-
-    options = jQuery.extend({
-
-        html:       "",
-        basePath:   false,
-        wymPath:    false,
-        iframeBasePath: false,
-        jQueryPath: false,
-        skin:       "default",
-        lang:       "en",
-        direction:  "ltr",
-        customCommands: [],
-
-        // These values will override the default for their respective
-        // `DocumentStructureManager.structureRules`
-        structureRules: {
-            defaultRootContainer:  'p'
-        },
+jQuery.fn.wymeditor = function (providedOptions) {
+    var $textareas = this,
+        mergedOptions,
+        optionsWithDefaults,
+        optionsWithSkinOverrides,
+        skin;
+
+
+    optionsWithDefaults = jQuery.extend({}, WYMeditor.DEFAULT_OPTIONS, providedOptions);
```

Comment:
We have four `jQuery.extend`s and:

* `optionsWithDefaults`
* `optionsWithSkinOverrides`
* `mergedOptions`

And `mergedOptions` is merged into twice.

Consistency would look better, in my taste.

How about we drop `optionsWithDefaults` and `optionsWithSkinOverrides`?